### PR TITLE
Integrate Core App into controller

### PR DESCRIPTION
**Pull request by grondinjc**
The solution is not fully integrated since it needs to be integrated into frontend

### CODE REVIEW

**Comment by grondinjc** [comment]
Async callback are not added yet. I'm working on this.

**Comment by grondinjc** [comment]
@vphilippon Check this for feedbacks please
(no need to merge, yet)

File: src/app/python/core.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+from threadpool import (ThreadPool,
+                        makeRequests as create_task)
+
+from libZoneTransit import (ZoneTransit, 
+                            Ajout as EditAdd, 
+                            Suppression as EditRemove)
+
+# Class to hold every change element using a name
+# Data field would be the conent if is_add is True and the count when False
```
Comment (maym2104) [inline src/app/python/core.py]: The count being the size of the deletion?

Comment (grondinjc) [inline src/app/python/core.py]: Yes

File: src/app/python/core.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    """
+    Core initialiser
+
+    @type project_name: str
+    @type project_path: str
+    @type logger: logging.Logger
+    @type num_threads: int
+
+    @param project_name: The CIDE.py logger instance
+    @param project_path: The CIDE.py logger instance
```
Comment (vphilippon) [inline src/app/python/core.py]: Change de params description.
Little copy-paste error.

File: src/server/ideController.py [hunk 3]
```diff
@@ -5,25 +6,54 @@
...
   """
 
-  def __init__(self, template_path, logger):
+  IDE_HTML_TEMPLATE = 'edit.html'
+
+  def __init__(self, app, template_path, logger):
     """
     IDEController initialiser
 
+    @param app: The core application
```
Comment (vphilippon) [inline src/server/ideController.py:47]: Add the `@type cide.app.python.core` type in docstring if possible. That's nice for people using autocompletion (like me!)

**Comment by vphilippon** [comment]
@grondinjc Pretty nice, added a few docsting notes.

File: bin/startCIDE.py [hunk 4]
```diff
@@ -46,10 +47,11 @@
 
 # Instanciate App
 chatApp = Chat(logger)
+coreApp = Core("dummyProjectName", "/tmp/test/dummyProjectName", logger, 4)
```
Comment (grondinjc) [inline bin/startCIDE.py:50]: @vphilippon should I use a different logger instance ?

Comment (vphilippon) [inline bin/startCIDE.py:50]: @grondinjc Not really, since we'll probably use the same config and file.

**Comment by grondinjc** [comment]
@vphilippon ready for review

File: bin/startCIDE.py (hunk 4 above)
Comment (vphilippon) [inline bin/startCIDE.py:50]: @grondinjc The project name should be set in a config file (May be done in next version)

File: src/server/ideController.py [hunk 5]
```diff
@@ -6,26 +7,101 @@
...
+
+    @param path: The object to validate as a path
+    """
+    return ((type(path) in (str, unicode)) and
+            (path == os.path.normpath(path.strip() or '/')) and 
+            (path.startswith('/')) and
+            (not path.endswith('/')))
+
+  @staticmethod
+  def is_valid_changes(changes):
```
Comment (vphilippon) [inline src/server/ideController.py:80]: Is it really needed? Let's think about it

File: src/server/ideController.py [hunk 6]
```diff
@@ -6,26 +7,101 @@
 import uuid  # XXX Temp for fake session id... Could be used for real?
 
 
+# Will be changed to httpErrors
+def create_argument_error_msg(arg):
+  return {'code': 400,
```
Comment (vphilippon) [inline src/server/ideController.py:12]: Use `cherrypy.response.status` or `raise HttpError` instead

**Comment by vphilippon** [comment]
There are a few things we'll want to change latter, but for the sake of going forward: :+1:

**Pull request merged**